File: Clase9/Agenda_1.java
/* 🧩 Descripción del Proyecto python/java
Vamos a crear una Agenda de Contactos, donde puedas:

1. Agregar un nuevo contacto (nombre, teléfono, correo). Clase contacto /Constructor

2. Mostrar todos los contactos guardados. Imprimir todos los atributos

3. Buscar contacto por nombre. Crear un menu con un while true con opción 
En python usar if_else y en java un swich/case

4. Eliminar un contacto.

5. Salir del programa.*/
package Clase9;

import java.util.ArrayList;
import java.util.HashMap;
import java.util.Map;
import java.util.Scanner;

public class Agenda_1 {
    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        ArrayList<Map<String, String>> contactos = new ArrayList<>();

        while (true) {
            mostrarMenu();
            int opcion = Integer.parseInt(sc.nextLine());

            switch (opcion) {
                case 1:
                    agregarContacto(sc, contactos);
                    break;
                case 2:
                    listarContactos(contactos);
                    break;
                case 3:
                    buscarPorNombre(sc, contactos);
                    break;
                case 4:
                    eliminarContacto(sc, contactos);
                    break;
                case 5:
                    System.out.println("Salir.");
                    return;
                


                default:
                    System.out.println("Opción inválida.");
            }
        }
    }

    // Mostrar el menú de opciones
    public static void mostrarMenu() {
        System.out.println("\n--- AGENDA PERSONAL ---");
        System.out.println("1. Agregar contacto");
        System.out.println("2. Listar contactos");
        System.out.println("3. Buscar por nombre");
        System.out.println("4. Eliminar");
        System.out.println("5. Salir");
        System.out.print("Seleccione una opción: ");
    }

    // Agregar un nuevo contacto
    public static void agregarContacto(Scanner sc, ArrayList<Map<String, String>> contactos) {
        Map<String, String> contacto = new HashMap<>();

        System.out.print("Nombre: ");
        String nombre = sc.nextLine();
        System.out.print("telefono: ");
        //int telefono = Integer.valueOf(sc.nextLine());
        String telefono = sc.nextLine();
        System.out.print("Email: ");
        String email = sc.nextLine();
       

        contacto.put("Nombre", nombre);
        contacto.put("Telefono", String.valueOf(telefono));
        contacto.put("Email", email);
        
        contactos.add(contacto);
        System.out.println("Contacto agregado exitosamente.");
    }

    
    // Mostrar todos los contactos
    public static void listarContactos(ArrayList<Map<String, String>> contactos) {
        if (contactos.isEmpty()) {
            System.out.println("No hay contactos registrados.");
            return;
        }

        System.out.println("\n--- Lista de Contactos ---");
        for (Map<String, String> contacto : contactos) {
            mostrarContacto(contacto);
        }
    }

    // Buscar contacto por nombre
    public static void buscarPorNombre(Scanner sc, ArrayList<Map<String, String>> contactos) {
        System.out.print("Ingrese el nombre a buscar: ");
        String nombreBuscado = sc.nextLine();
        boolean encontrado = false;

        for (Map<String, String> contacto : contactos) {
            if (contacto.get("Nombre").equalsIgnoreCase(nombreBuscado)) {
                mostrarContacto(contacto);
                encontrado = true;
            }
        }

        if (!encontrado) {
            System.out.println("Contacto no encontrado.");
        }
    }

        public static void eliminarContacto(Scanner sc, ArrayList<Map<String, String>> contactos) {
        System.out.print("Ingrese el nombre del contacto a eliminar: ");
        String nombreEliminar = sc.nextLine();
        boolean eliminado = false;

        for (int i = 0; i < contactos.size(); i++) {
            if (contactos.get(i).get("Nombre").equalsIgnoreCase(nombreEliminar)) {
                contactos.remove(i);
                eliminado = true;
                System.out.println("Contacto eliminado exitosamente.");
                break;
            }
        }

        if (!eliminado) {
            System.out.println("No se encontró un contacto con ese nombre.");
        }
    }


    // Mostrar un contacto individual
    public static void mostrarContacto(Map<String, String> contacto) {
        System.out.println("----------------------------");
        for (Map.Entry<String, String> entry : contacto.entrySet()) {
            System.out.println(entry.getKey() + ": " + entry.getValue());
        }
    }
    

}
/*              case "4":
                    System.out.print("Nombre a eliminar: ");
                    String deleteName = sc.nextLine();
                    Iterator<Contact> iterator = contacts.iterator();
                    while (iterator.hasNext()) {
                        Contact c = iterator.next();
                        if (c.name.equalsIgnoreCase(deleteName)) {
                            iterator.remove();
                        }
                    }
                    break;

                case "5":
                    System.out.println("¡Adiós!");
                    return;

                default:
                    System.out.println("Opción no válida");
                    break; */

File: Clase9/Agenda_1.py
agenda = {}

def agregar_contacto():
    nombre = input("Ingrese el nombre del contacto: ")
    telefono = input("Ingrese el número de teléfono: ")
    correo = input("Ingrese el correo electrónico: ")
    agenda[nombre] = {"Teléfono": telefono, "Correo": correo}  ##Esto no se me occurrio y se lo pregunte a Chat
    print(f"Contacto '{nombre}' agregado exitosamente.\n")  ##Esto no se me ocurrio pero me parece un buen validador chat gpt

def mostrar_contactos():
    if not agenda:
        print("No hay contactos guardados.\n")
    else:
        print("\n--- Lista de Contactos ---")
        for nombre, info in agenda.items():
            print(f"Nombre: {nombre}")
            print(f"  Teléfono: {info['Teléfono']}")
            print(f"  Correo: {info['Correo']}\n")

def buscar_contacto():
    nombre = input("Ingrese el nombre del contacto a buscar: ")
    if nombre in agenda:
        print(f"\nContacto encontrado:")
        print(f"Nombre: {nombre}")
        print(f"  Teléfono: {agenda[nombre]['Teléfono']}")
        print(f"  Correo: {agenda[nombre]['Correo']}\n")
    else:
        print(f"El contacto '{nombre}' no se encuentra en la agenda.\n")

def eliminar_contacto():
    nombre = input("Ingrese el nombre del contacto a eliminar: ")
    if nombre in agenda:
        del agenda[nombre]
        print(f"Contacto '{nombre}' eliminado exitosamente.\n")
    else:
        print(f"El contacto '{nombre}' no existe en la agenda.\n")

def menu():
    while True:
        print("=== Agenda de Contactos ===")
        print("1. Agregar un nuevo contacto")
        print("2. Mostrar todos los contactos")
        print("3. Buscar contacto por nombre")
        print("4. Eliminar un contacto")
        print("5. Salir")
        
        opcion = input("Seleccione una opción (1-5): ")
        
        if opcion == '1':
            agregar_contacto()
        elif opcion == '2':
            mostrar_contactos()
        elif opcion == '3':
            buscar_contacto()
        elif opcion == '4':
            eliminar_contacto()
        elif opcion == '5':
            print("Saliendo del programa. ¡Hasta luego!")
            break
        else:
            print("Opción no válida. Intente nuevamente.\n")

menu()
